sparse/src/KokkosSparse_spgemm_handle.hpp

```diff
@@ -148,45 +148,81 @@ class SPGEMMHandle {
   typedef typename nnz_lno_persistent_work_view_t::HostMirror
       nnz_lno_persistent_work_host_view_t;  // Host view type
 
-#ifdef KOKKOSKERNELS_ENABLE_TPL_ROCSPARSE
-  struct rocSparseSpgemmHandleType {
-    KokkosKernels::Experimental::Controls
-        kkControls;  // give a singleton rocsparse handle
-    rocsparse_handle rocsparseHandle;
-    rocsparse_operation opA, opB;
-    rocsparse_mat_descr descr_A, descr_B, descr_C, descr_D;
-    rocsparse_mat_info info_C;
-    size_t bufferSize;
-    void *buffer;
-    bool C_populated;
+#if defined(KOKKOSKERNELS_ENABLE_TPL_ROCSPARSE)
+
+  struct rocSparseSpgemmHandleType
+  {
+
+    /// Constructor
+    rocSparseSpgemmHandleType():
```

Comment:
The API of Kokkos Kernels cannot arbitrarily change, you need to deprecate the old constructor until the next major release. You also need to motivate why your new implementation is more appropriate to users than the current one.